includes/express-checkout/class-wc-payments-express-checkout-button-handler.php

```diff
@@ -257,6 +257,15 @@ public function scripts() {
 			'is_checkout_page'   => $this->express_checkout_helper->is_checkout(),
 		];
 
+		// No need to send a list of countries if the store:
+		// 1. ships to all countries. Regardless of whom it sells to.
+		$ship_to_all = get_option( 'woocommerce_ship_to_countries' ) === 'all';
+		// 2. ships to all countries it sells to and store sells to all countries.
+		$ship_to_all_and_selling_to_all = get_option( 'woocommerce_allowed_countries' ) === 'all' && empty( get_option( 'woocommerce_ship_to_countries' ) );
+		if ( ! $ship_to_all && ! $ship_to_all_and_selling_to_all ) {
```

Comment:
There’s no real need to add a condition here. I’m only doing this to be extra cautious in case Stripe supports a country code that WooPayments (or rather WooCommerce) doesn’t include in its list. For example, if Stripe supports the country code ‘ZZ’, but WooPayments doesn’t, the best way to handle ‘ZZ’ is to send no country list and let Stripe handle all countries (which is the default behavior when no list is provided)

Reply:
I wanted to double check the optimization I'm doing here and found a discrepancy with this condition. The behavior changes when the condition is removed, which I didn’t expect.

If `allowed_shipping_countries` is set to the list of all countries provided by WooCommerce, ECE displays a warning in the browser console, indicating that those are invalid country codes:

```
- AS: American Samoa
- PW: Belau
- CX: Christmas Island
- CC: Cocos (Keeling) Islands
- CU: Cuba
- HM: Heard Island and McDonald Islands
- IR: Iran
- MH: Marshall Islands
- FM: Micronesia
- NF: Norfolk Island
- KP: North Korea
- MP: Northern Mariana Islands
- SY: Syria
- UM: United States (US) Minor Outlying Islands
- VI: Virgin Islands (US)
```

Warnings:

```
controller-843927f4a5c13aa00b34dffe4221429a.js:1 [Stripe.js] invalid country code "AS"; "AS" is not a valid 2-letter country code
warn @ controller-843927f4a5c13aa00b34dffe4221429a.js:1
warn @ controller-843927f4a5c13aa00b34dffe4221429a.js:1
(anonymous) @ controller-843927f4a5c13aa00b34dffe4221429a.js:1
controller-843927f4a5c13aa00b34dffe4221429a.js:1 [Stripe.js] invalid country code "PW"; "PW" is not a valid 2-letter country code
warn @ controller-843927f4a5c13aa00b34dffe4221429a.js:1
warn @ controller-843927f4a5c13aa00b34dffe4221429a.js:1
...
```

and the payment sheet doesn't allow selecting a shipping address from those countries. However, when the condition is added and we avoid passing a list, leaving ECE to support all countries, it does allow checkout with addresses from those countries. So far, I’ve tested with AS (American Samoa) and PW (Belau or Palau) and it was able to checkout without problems.

For reference, I haven’t been able to define addresses for some countries like Cuba or Iran. I assume if Google Pay allowed me to add these addresses, ECE would allow me to checkout using a shipping address from those countries.

@bborman22 @pierorocca what should we do in this case? Should we keep the condition or remove it? If we keep the condition, we address the potential bug mentioned earlier, but it prevents shipping addresses from the countries listed above. If we remove the condition, we allow shipping to those countries via ECE.

Stripe’s behavior seems inconsistent. I'd expect that if any of these country codes are restricted or not allowed, they wouldn't be permitted by default when no countries are passed to `allowedShippingCountries`.

Reply:
I've removed the condition, as it seems the safest option to honor the restrictions. Stripe should address the issue where restricted countries are allowed for shipping when `allowedShippingCountries` is not defined.